src/game/Server/WorldSession.cpp

```diff
@@ -516,8 +578,11 @@ void WorldSession::LogoutPlayer(bool Save)
         // No SQL injection as AccountId is uint32
 
         static SqlStatementID updChars;
-
+#ifdef ENABLE_PLAYERBOTS
+        SqlStatement stmt = CharacterDatabase.CreateStatement(updChars, "UPDATE characters SET online = 0 WHERE account = ?");
+#else
         stmt = CharacterDatabase.CreateStatement(updChars, "UPDATE characters SET online = 0 WHERE account = ?");
```

Comment:
In my mind the code changing like this if defines change feels bad.
For maintainability remove the ifdefs and the normal SQL statement so only thing left is this:
```
SqlStatement stmt = CharacterDatabase.CreateStatement(updChars, "UPDATE characters SET online = 0 WHERE account = ?");
```

Then to solve the issue with the variable names (redefinition of stmt) wrap the query from before in brackets like so:
```
{
    SqlStatement stmt = LoginDatabase.CreateStatement(id, "UPDATE account SET active_realm_id = ? WHERE id = ?");
    stmt.PExecute(uint32(0), GetAccountId());
}
```
**_Or alternatively use different names for stmt variables at different points which also avoids the whole issue._**